bridge/tracker/tracker.go

```diff
@@ -0,0 +1,390 @@
+//	between blocks fromBlock and toBlock (inclusive).
+func (t *Tracker) queryEvents(fromBlock, toBlock uint64) []*web3.Log {
+	queryFilter := &web3.LogFilter{}
+
+	//	set range of blocks to query
+	queryFilter.SetFromUint64(fromBlock)
+	queryFilter.SetToUint64(toBlock)
+
+	//	set smart contract addresses
+	queryFilter.Address = []web3.Address{
+		web3.HexToAddress(PoCSC),
+		web3.HexToAddress(AnotherEventSC),
+		web3.HexToAddress(ThirdEventSC),
+	}
+
+	//	set relevant topics (defined in contracts)
+	queryFilter.Topics = [][]*web3.Hash{
+		{
+			&topicPoCEvent,
+			&topicAnotherEvent,
+		},
+	}
+
+	//	call eth_getLogs
+	logs, err := t.client.getLogs(queryFilter)
+	if err != nil {
+		t.logger.Error("eth_getLogs failed", "err", err)
+
+		return nil
+	}
+
+	//	overwrite checkpoint
+	if err := t.saveLastBlock(big.NewInt(0).SetUint64(toBlock)); err != nil {
+		t.logger.Error("cannot save last block number proccesed", "err", err)
+	}
+
+	return logs
+}
+
+//	notify sends the given logs to the event channel.
+func (t *Tracker) notify(logs ...*web3.Log) {
+	for _, log := range logs {
+		bytesLog, err := json.Marshal(log)
+		if err != nil {
+			t.logger.Error("cannot marshal log", "err", err)
+		}
+
+		// notify
+		select {
+		case t.eventCh <- bytesLog:
+		default:
+		}
+	}
+}
+
+/* Header structs parsed by the client */
+
+//	Ethereum header
+//type ethHeader struct {
+//	Difficulty   string        `json:"difficulty"`
+//	ExtraData    string        `json:"extraData"`
+//	GasLimit     string        `json:"gasLimit"`
+//	GasUsed      string        `json:"gasUsed"`
+//	LogsBloom    types.Bloom   `json:"logsBloom"`
+//	Miner        types.Address `json:"miner"`
+//	Nonce        string        `json:"nonce"`
+//	Number       string        `json:"number"`
+//	ParentHash   types.Hash    `json:"parentHash"`
+//	ReceiptsRoot types.Hash    `json:"receiptsRoot"`
+//	Sha3Uncles   types.Hash    `json:"sha3Uncles"`
+//	StateRoot    types.Hash    `json:"stateRoot"`
+//	Timestamp    string        `json:"timestamp"`
+//	TxRoot       types.Hash    `json:"transactionsRoot"`
+//	MixHash      types.Hash    `json:"mixHash"`
+//	Hash         types.Hash    `json:"hash"`
+//}
+
+//	Polygon-Edge header
+type ethHeader struct {
+	Difficulty   uint64        `json:"difficulty"`
+	ExtraData    string        `json:"extraData"`
+	GasLimit     uint64        `json:"gasLimit"`
+	GasUsed      uint64        `json:"gasUsed"`
+	LogsBloom    types.Bloom   `json:"logsBloom"`
+	Miner        types.Address `json:"miner"`
+	Nonce        string        `json:"nonce"`
+	Number       uint64        `json:"number"`
+	ParentHash   types.Hash    `json:"parentHash"`
+	ReceiptsRoot types.Hash    `json:"receiptsRoot"`
+	Sha3Uncles   types.Hash    `json:"sha3Uncles"`
+	StateRoot    types.Hash    `json:"stateRoot"`
+	Timestamp    uint64        `json:"timestamp"`
+	TxRoot       types.Hash    `json:"transactionsRoot"`
+	MixHash      types.Hash    `json:"mixHash"`
+	Hash         types.Hash    `json:"hash"`
+}
```

Comment:
What do you think about moving this to a separate file?

Reply:
[Moved](https://github.com/0xPolygon/polygon-edge/pull/386/commits/44989ee14dbef25e79c23bca1793f88427a372cd).

Moved to `rootchain.go`, since it makes most sense. We can discuss how to better structure that file.